extension/src-language-server/stpa/diagram/diagram-generator.ts

```diff
@@ -872,38 +1105,49 @@ export class StpaDiagramGenerator extends LangiumDiagramGenerator {
         targetId: string,
         label: string[],
         edgeType: EdgeType,
-        args: GeneratorContext<Model>
+        sedgeType: string,
+        args: GeneratorContext<Model>,
+        dummyLabel: boolean = true
     ): CSEdge {
         return {
-            type: CS_EDGE_TYPE,
+            type: sedgeType,
             id: edgeId,
             sourceId: sourceId!,
             targetId: targetId!,
             edgeType: edgeType,
-            children: this.createLabel(label, edgeId, args.idCache),
+            children: this.createLabel(label, edgeId, args.idCache, undefined, dummyLabel),
         };
     }
 
     /**
      * Generates SLabel elements for the given {@code label}.
      * @param label Labels to translate to SLabel elements.
      * @param id The ID of the element for which the label should be generated.
+     * @param idCache The ID cache of the STPA model.
+     * @param type The type of the label.
+     * @param dummyLabel Determines whether a dummy label should be created to get a correct layout.
      * @returns SLabel elements representing {@code label}.
      */
-    protected createLabel(label: string[], id: string, idCache: IdCache<AstNode>): SLabel[] {
+    protected createLabel(
+        label: string[],
+        id: string,
+        idCache: IdCache<AstNode>,
+        type: string = "label:xref",
+        dummyLabel: boolean = true
+    ): SLabel[] {
         const children: SLabel[] = [];
-        if (label.find((l) => l !== "")) {
-            label.forEach((l) => {
+        if (label.find(l => l !== "")) {
+            label.forEach(l => {
                 children.push({
-                    type: "label:xref",
+                    type: type,
                     id: idCache.uniqueId(id + "_label"),
                     text: l,
                 } as SLabel);
             });
-        } else {
+        } else if (dummyLabel) {
             // needed for correct layout
             children.push({
-                type: "label:xref",
+                type: type,
```

Comment:
this file has become quite large and bloated, in the future a refactor might be in order